lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVMBase.h

```diff
@@ -879,24 +888,57 @@ class ConvertTritonGPUOpToLLVMPatternBase {
     return multiDimIdx;
   }
 
+  SmallVector<SmallVector<unsigned>>
+  emitOffsetForSliceLayout(const SliceEncodingAttr &sliceLayout,
+                           RankedTensorType type) const {
+    auto parentEncoding = sliceLayout.getParent();
+    unsigned dim = sliceLayout.getDim();
+    auto parentShape = sliceLayout.paddedShape(type.getShape());
+    RankedTensorType parentTy = RankedTensorType::get(
+        parentShape, type.getElementType(), parentEncoding);
+    auto parentOffsets = emitOffsetForLayout(parentEncoding, parentTy);
+
+    unsigned numOffsets = parentOffsets.size();
+    SmallVector<SmallVector<unsigned>> resultOffsets;
+    std::set<SmallVector<unsigned>> uniqueOffsets;
+
+    for (unsigned i = 0; i < numOffsets; ++i) {
+      SmallVector<unsigned> offsets = parentOffsets[i];
+      offsets.erase(offsets.begin() + dim);
+      if (uniqueOffsets.find(offsets) == uniqueOffsets.end()) {
+        resultOffsets.push_back(offsets);
+        uniqueOffsets.insert(offsets);
+      }
+    }
+    return resultOffsets;
+  }
+
   SmallVector<SmallVector<Value>>
   emitIndicesForSliceLayout(Location loc, ConversionPatternRewriter &rewriter,
                             const SliceEncodingAttr &sliceLayout,
                             RankedTensorType type) const {
     auto parentEncoding = sliceLayout.getParent();
-    unsigned dim = sliceLayout.getDim();
     auto parentShape = sliceLayout.paddedShape(type.getShape());
     RankedTensorType parentTy = RankedTensorType::get(
         parentShape, type.getElementType(), parentEncoding);
-    auto parentIndices = emitIndices(loc, rewriter, parentEncoding, parentTy);
-    unsigned numIndices = parentIndices.size();
-    SmallVector<SmallVector<Value>> resultIndices;
-    for (unsigned i = 0; i < numIndices; ++i) {
-      SmallVector<Value> indices = parentIndices[i];
-      indices.erase(indices.begin() + dim);
-      resultIndices.push_back(indices);
-    }
-    return resultIndices;
+
+    unsigned dim = sliceLayout.getDim();
```

Comment:
This segment almost looks like a copy-paste of emitIndicesForDistributedLayout. Any way this could be shared between both code paths?

Reply:
Great catch! They ended up being exactly the same.